app/src/main/java/com/framgia/wsm/widget/holidaycalendar/HolidayCalendarView.java

```diff
@@ -0,0 +1,368 @@
+package com.framgia.wsm.widget.holidaycalendar;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.content.res.TypedArray;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.Typeface;
+import android.support.annotation.Nullable;
+import android.text.format.Time;
+import android.util.AttributeSet;
+import android.view.MotionEvent;
+import android.view.View;
+import com.framgia.wsm.R;
+import com.framgia.wsm.data.model.HolidayCalendarDate;
+import com.framgia.wsm.utils.common.DateTimeUtils;
+import java.text.DateFormatSymbols;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+public class HolidayCalendarView extends View {
+
+    private static final int SELECTED_CIRCLE_ALPHA = 128;
+    private static final int DEFAULT_HEIGHT = 32;
+    private int mDaySelectedCircleSize;
+    private int mDaySeparatorWidth = 1;
+    private int mMiniDayNumberTextSize;
+    private int mMonthDayLabelTextSize;
+    private int mMonthHeaderSize;
+    private int mMonthLabelTextSize;
+    private int mNumRows;
+
+    private int mPadding = 0;
+
+    private Paint mMonthDayLabelPaint;
+    private Paint mMonthNumPaint;
+    private Paint mReservationInquiryStatusPaint;
+    private Paint mMonthTitlePaint;
+    private Paint mSelectedCirclePaint;
+    private int mMonthTextColor;
+    private int mDayTextColor;
+    private int mSaturdaySundayColor;
+    private int mCompanyHoliday;
+    private int mNormalHoliday;
+
+    private StringBuilder mStringBuilder;
```

Comment:
WARNING: [AvoidStringBufferField] StringBuffers can grow quite a lot, and so may become a source of memory leak (if the owning class has a long life time).
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/strings.html#AvoidStringBufferField